chain/rpc/src/client.rs

```diff
@@ -152,435 +176,419 @@ pub struct SimpleJsonRpcRetryPolicy {
     pub max_retry_queue_size: u32,
 }
 
-impl SimpleJsonRpcRetryPolicy {
-    fn is_retryable_json_rpc_error(&self, err: &JsonRpcError) -> bool {
-        self.retryable_json_rpc_errors.contains(&err.code) || err.message.contains("rate limit")
+impl DefaultRetryPolicy {
+    fn is_retryable_json_rpc_errors(&self, rpc_err: &ErrorPayload) -> bool {
+        self.retryable_json_rpc_errors.contains(&rpc_err.code)
     }
 
-    fn is_retryable_http_error(&self, status: &http::StatusCode) -> bool {
-        self.retryable_http_errors.contains(status)
+    fn is_retryable_http_errors(&self, http_err: &HttpError) -> bool {
+        let status_code = match http::StatusCode::try_from(http_err.status) {
+            Ok(status_code) => status_code,
+            Err(_) => return false,
+        };
+        self.retryable_http_errors.contains(&status_code)
     }
 }
 
-impl RetryPolicy<JsonRpcProviderClientError> for SimpleJsonRpcRetryPolicy {
-    fn is_retryable_error(
-        &self,
-        err: &JsonRpcProviderClientError,
-        num_retries: u32,
-        retry_queue_size: u32,
-    ) -> RetryAction {
-        if self.max_retries.is_some_and(|max| num_retries > max) {
-            warn!(
-                count = self.max_retries.expect("max_retries must be set"),
-                "max number of retries has been reached"
-            );
-            return NoRetry;
-        }
+impl RetryPolicy for DefaultRetryPolicy {
+    // TODO: original implementation requires input param of `num_retries`
```

Comment:
Is this TODO resolved?

Reply:
Same. Related to issue https://github.com/hoprnet/hoprnet/issues/7139